addons/stock/migrations/13.0.1.1/post-migration.py

```diff
@@ -0,0 +1,364 @@
+        FROM stock_quant_package sqp
+        WHERE sml.package_id = sqp.id AND sml.company_id IS NULL"""
+    )
+    # stock.picking.type
+    openupgrade.logged_query(
+        cr, """
+        UPDATE stock_picking_type spt
+        SET company_id = sw.company_id
+        FROM stock_warehouse sw
+        WHERE spt.warehouse_id = sw.id AND spt.company_id IS NULL"""
+    )
+    # stock.picking
+    openupgrade.logged_query(
+        cr, """
+        UPDATE stock_picking sp
+        SET company_id = spt.company_id
+        FROM stock_picking_type spt
+        WHERE sp.picking_type_id = spt.id AND sp.company_id IS NULL"""
+    )
+    # stock.package_level
+    openupgrade.logged_query(
+        cr, """
+        UPDATE stock_package_level spl
+        SET company_id = sqp.company_id
+        FROM stock_quant_package sqp
+        WHERE spl.package_id = sqp.id AND spl.company_id IS NULL"""
+    )
+    # stock.scrap
+    openupgrade.logged_query(
+        cr, """
+        UPDATE stock_scrap ss
+        SET company_id = COALESCE(
+            (SELECT sm.company_id FROM stock_move sm
+             WHERE sm.id = ss.move_id AND sm.company_id IS NOT NULL
+             LIMIT 1), ru.company_id)
+        FROM res_users ru
+        WHERE ru.id = ss.create_uid AND ss.company_id IS NULL"""
+    )
+    # stock.putaway.rule
+    openupgrade.logged_query(
+        cr, """
+        UPDATE stock_putaway_rule spr
+        SET company_id = ru.company_id
+        FROM res_users ru
+        WHERE ru.id = spr.create_uid AND spr.company_id IS NULL"""
+    )
+    # stock.production.lot
+    openupgrade.logged_query(
+        cr, """
+        UPDATE stock_production_lot spl
+        SET company_id = COALESCE(
+            (SELECT sml.company_id FROM stock_move_line sml
+             WHERE sml.lot_id = spl.id AND sml.company_id IS NOT NULL
+             LIMIT 1), ru.company_id)
+        FROM res_users ru
+        WHERE ru.id = spl.create_uid AND spl.company_id IS NULL"""
+    )
+
+
+def fill_stock_putaway_rule_location_in_id(env):
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE stock_putaway_rule spr
+        SET location_in_id = sl.id
+        FROM stock_location sl
+        JOIN product_putaway pp ON sl.putaway_strategy_id = pp.id
+        WHERE spr.putaway_id = pp.id AND spr.location_in_id IS NULL"""
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        INSERT INTO stock_putaway_rule (product_id, category_id,
+            location_in_id, location_out_id, sequence, company_id,
+            create_uid, create_date, write_uid, write_date)
+        SELECT spr.product_id, spr.category_id, sl.id, spr.location_out_id,
+            spr.sequence, spr.company_id, spr.create_uid, spr.create_date,
+            spr.write_uid, spr.write_date
+        FROM stock_putaway_rule spr
+        JOIN product_putaway pp ON spr.putaway_id = pp.id
+        JOIN stock_location sl ON sl.putaway_strategy_id = pp.id
+        WHERE spr.location_in_id != sl.id"""
```

Comment:
<pre>
FROM stock_putaway_rule spr
  JOIN stock_location sl
    ON sl.putaway_strategy_id = spr.putaway_id and spr.location_in_id != sl.id
</pre>